app/assets/javascript/app/components/SearchBar.jsx

```diff
@@ -1,14 +1,91 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+import Tags from './Tags';
+
 
 class SearchBar extends React.Component {
 
+  constructor(props) {
+    super(props);
+
+    this.state = { 
+      query: '',
+    };
+
+    this.timer = null;
+    this.debounce = 500;
+
+    this.setQuery = this.setQuery.bind(this);
+    this.clearQuery = this.clearQuery.bind(this);
+  }
+
+
+  setQuery(e) {
+    const query = e.target.value;
+
+    if (this.timer) {
+      clearTimeout(this.timer);
+    }
+
+    this.timer = setTimeout(() => {
+      this.props.fetchResults(query);
+    }, this.debounce);
+
+    this.setState({ query });
+  }
+
+
+  clearQuery() {
+    this.setState({ query: '' });
+  }
+
+
   render() {
+    const results = this.props.results.map(result => {
+      const classes = result.type === 'permit' ? 'borderless' : '';
+      const styles = { borderColor: result.color };
+
+      const diamond = (() => {
```

Comment:
Does this really need to be scrap scoped? Seems like this could be a quick ternary.